yarn-project/foundation/src/timer/timeout.ts

```diff
@@ -10,14 +12,9 @@ export class TimeoutTask<T> {
   private interrupt = () => {};
   private totalTime = 0;
 
-  constructor(
-    private fn: () => Promise<T>,
-    private timeout = 0,
-    fnName = '',
-    error = () => new Error(`Timeout${fnName ? ` running ${fnName}` : ''} after ${timeout}ms.`),
-  ) {
+  constructor(private fn: () => Promise<T>, private timeout: number, errorFn: () => any) {
     this.interruptPromise = new Promise<T>((_, reject) => {
-      this.interrupt = () => reject(error());
+      this.interrupt = () => reject(errorFn());
```

Comment:
Not introduced in this PR. But this class scares me a bit. If an operation is interrupted, doesn't it leave the operation conitnuing? Which if it rejects results in an unhandled promise rejection? If you agree, I think we should tackle this in a follow up.

Reply:
It seems `Promise.race` deliberately ignores rejections from all promises but the first one, so we're good.

See https://stackoverflow.com/a/70403594